### ISSUE TITLE
should reconstruct function be forward pass only?

### ISSUE THREAD

**Issue by oafolabi** [facebookresearch/DeepSDF#9]
Hi, 

I'm trying to understand why the reconstruct function [here](https://github.com/facebookresearch/DeepSDF/blob/a55416a995ae1c918f18e32681d262853b496788/reconstruct.py#L16) isn't just a forward pass.  Why is there a backward call [here](https://github.com/facebookresearch/DeepSDF/blob/a55416a995ae1c918f18e32681d262853b496788/reconstruct.py#L75)?

Thanks!

**Comment by nicolasugrinovic** [comment]
@oafolabi, (I did no build the system so don't take my word for granted) 
I believe it is because the reconstruction is framed as an optimization. Given the weights learned in the training face, here backwards and value update: 
`        loss.backward()
        optimizer.step()`
only updates the latent vector z values and not the network parameters.

**Comment by JiamingSuen** [comment]
> Yet, maybe the question remains, why use GT at reconstruction?

+1 for this question. @tschmidt23 Hope you can clarify this for us, thanks!

**Comment by tschmidt23** [comment]
@nicolasugrinovic is correct, inferring a shape code involves using gradient descent to find a code that (locally) minimizes prediction error for a set of observed SDF values. The backward pass through the network computes the gradient of the prediction error w.r.t. the shape code. See eq. 10 in the paper.

**Comment by nicolasugrinovic** [comment]
@tschmidt23, thank you for the clarification. It makes sense. You want to use "GT" because that is what you want to represent. Clever way of reusing parameters! Little fuzzy to grasp because it is not very _status quo_

**Comment by moshanATucsd** [comment]
Hi, for eq. 10, does it mean during inference we still need to input some observed SDF values? Can we just input point clouds?
